tests/test_data.py

```diff
@@ -151,6 +156,84 @@ def test_collection_get(self):
         # This is unacceptable
         self.assertRaises(ValueError, locations.get, ("37.538758 N", "-77.433594 W"))
 
+    def test_dict_collection(self):
+        """Ensure that the LocationCollectionDict stores as a dict of Locations"""
+        self.assertTrue(isinstance(self.batch_dict_response, dict))
+        locations = LocationCollectionDict(self.batch_dict_response["results"])
+
+        self.assertTrue(isinstance(locations["1"], Location))
+
+
+    def test_dict_collection_coords(self):
+        """Ensure the coords property returns a list of suitable tuples"""
+        locations = LocationCollectionDict(self.batch_dict_response["results"])
+        self.assertEqual(
+            locations.coords,
+            {
+                "1":(37.560890255102, -77.477400571429),
+                "2":(37.554895702703, -77.457561054054),
+                "3":None
+            },
+        )
+
+        # Do the same with the order changed
+        locations = LocationCollectionDict(self.batch_dict_response["results"], order="lng")
+        self.assertEqual(
+            locations.coords,
+            {
+                "1":(-77.477400571429, 37.560890255102),
+                "2":(-77.457561054054, 37.554895702703),
+                "3":None
+            },
+        )
+
+    def test_dict_collection_addresses(self):
+        """Ensure that formatted addresses are returned"""
+        locations = LocationCollectionDict(self.batch_dict_response["results"])
+        self.assertEqual(
+            locations.formatted_addresses,
+            {
+                "1":"3101 Patterson Ave, Richmond VA, 23221",
+                "2":"1657 W Broad St, Richmond VA, 23220",
+                "3":"",
+            },
+        )
+
+    def test_dict_collection_get(self):
+        """Ensure 'get' performs a key based lookup"""
+        locations = LocationCollectionDict(self.batch_dict_response["results"])
+        self.assertEqual(
+            locations.get("3101 patterson ave, richmond, va").coords,
+            (37.560890255102, -77.477400571429),
+        )
+
+        """Ensure 'get' is able to look up by dictionary key"""
+        self.assertEqual(
+            locations.get("1").coords,
+            (37.560890255102, -77.477400571429),
+        )
+
+        # Case sensitive on the specific query
+        self.assertEqual(KeyError, locations.get("3101 Patterson Ave, richmond, va"))
```

Comment:
I had to update the existing `get` tests to `assertEquals` on `KeyError` from `assertRaises`, as the python dict `get` method avoids raising a `KeyError` by default.